src/operations/blas3/gemm_local.hpp

```diff
@@ -199,25 +186,27 @@ class Gemm<input_t, output_t, DoubleBuffer, NbcA, NbcB, ClSize, TileType,
    * group to multiple work groups with size as expected by GemmFactory::run().
    * (This is done by manipulating wg_id and item_id parameters.)
    */
-  static SYCL_BLAS_INLINE cl::sycl::nd_range<1> get_nd_range(
-      index_t m, index_t n, index_t compute_units) noexcept {
-    const cl::sycl::range<1> nwg(
-        get_workgroup_cluster(m, n) *
-        get_num_workgroup_cluster(m, n, compute_units));
+  SYCL_BLAS_INLINE cl::sycl::nd_range<1> get_nd_range(
+      index_t compute_units) const noexcept {
+    const cl::sycl::range<1> nwg(get_workgroup_cluster() *
+                                 get_num_workgroup_cluster(compute_units));
     const cl::sycl::range<1> wgs(wg_size);
 #ifdef VERBOSE
-    std::cout << " M: " << m << " , N " << n
+    std::cout << " M: " << a_.get_size_row() << " , N " << b_.get_size_col()
               << " , big_tile_rows: " << big_tile_rows
               << " , big_tile_cols: " << big_tile_cols
               << " , wg_size: " << wg_size << " , nwg : "
-              << ((m - 1) / big_tile_rows + 1) * ((n - 1) / big_tile_cols + 1) *
-                     tl_rows * tl_cols
+              << ((a_.get_size_row() - 1) / big_tile_rows + 1) *
```

Comment:
Is this the same as `get_workgroup_cluster()`? Would it make sense to use that here instead of repeating the computation?